### PR TITLE
Assert rev in pillow processor

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @snopoke this'll publish the doc rev to the changes feed and then assert that the revs are the same. curious to see if any of these pop up in the coming days. i think we only need it for the ElasticProcessor but i may be overlooking another place where it might be useful

http://manage.dimagi.com/default.asp?237983
http://manage.dimagi.com/default.asp?237943

cc: @proteusvacuum

### CODE REVIEW

File: corehq/apps/change_feed/tests/test_pillow.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # coding=utf-8
+from mock import patch, MagicMock
```
Comment (dimagimon) [inline corehq/apps/change_feed/tests/test_pillow.py]: F401 'MagicMock' imported but unused

**Comment by czue** [comment]
looks good and no blockers on my side. wondering if we should apply this more broadly? I can't think of any instance where you wouldn't want to fail hard on out-of-date revisions and it seems like this would cause problems in UCRs and custom reports and probably most other things we use reports for...

**Comment by benrudolph** [comment]
yep that sounds right to me

**Comment by benrudolph** [comment]
@czue cleaned up and added some more checks

**Review by czue — APPROVED** [review]
looks good on :white_check_mark:

**Pull request merged**